src/IonEvent.ts

```diff
@@ -395,20 +395,28 @@ class IonStructEvent extends AbsIonContainerEvent {//no embed support as of yet.
     }
     //two way equivalence between the structEvents, they must have the exact same number of equivalent elements.
     valueEquals(expected : IonStructEvent) : boolean {
-        return expected instanceof IonStructEvent && this.structsEqual(this.ionValue, expected.ionValue) && this.structsEqual(expected.ionValue, this.ionValue);
+        return expected instanceof IonStructEvent && this.ionValue.length === expected.ionValue.length && this.structsEqual(this.ionValue, expected.ionValue);
     }
 
     //for each actual ionEvent, searches for an equivalent expected ionEvent,
     //equivalent pairings remove the paired expectedEvent from the search space.
     structsEqual(actualEvents : AbstractIonEvent[], expectedEvents : AbstractIonEvent[]) : boolean {
         let matchFound : boolean = true;
         let paired : boolean[] = new Array<boolean>(expectedEvents.length);
-        for(let i : number = 0; matchFound && i < actualEvents.length; i++) {
+        for (let i : number = 0; matchFound && i < actualEvents.length; i++) {
             matchFound = false;
-            for(let j : number = 0; !matchFound && j < expectedEvents.length; j++) {
-                if(!paired[j]) {
-                    matchFound = actualEvents[i].equals(expectedEvents[j]);
-                    paired[j] = matchFound;
+            for (let j : number = 0; !matchFound && j < expectedEvents.length; j++) {
+                if (!paired[j]) {
+                    let child = actualEvents[i]
+                    let expectedChild = expectedEvents[j]
+                    matchFound = child.equals(expectedChild);
+                    if (matchFound) paired[j] = true;
+                    if (matchFound && child.eventType === IonEventType.CONTAINER_START) {
+                        for (let k = j + 1; k < expectedChild.ionValue.length; k++) {
+                            paired[k] = true;
+                        }
+                        i += child.ionValue.length;
+                    }
```

Comment:
So this is basically doing a bag comparison, yes?  A comment explaining what the intent of this algorithm is would be helpful.  Also, explaining why ES6 `Map` cannot be used to reduce this O(N**2) to O(N) would be helpful.

Reply:
it can be done in O(n)!